sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -498,100 +624,82 @@ enum class matrix_type {
   uint32,
   uint64
 };
-
-enum class scope_t {
-  sub_group,
-  work_group
+struct combination {
+  uint32_t max_msize;
+  uint32_t max_nsize;
+  uint32_t max_ksize;
+  uint32_t msize;
+  uint32_t nsize;
+  uint32_t ksize;
+  matrix_type atype;
+  matrix_type btype;
+  matrix_type accumulatortype;
 };
-}
-```
-
-
-=== Validation Example:
-```c++
-// User can provide sizes besides the types and tpu_params can assert if they are supported or not
-// in this case, an assertion will happens as 16 is not a supported size for M
-using myparams = tpu_params<tpu::xmx16, int8_t, int8_t, int, 16, 16, 32>;  
-size_t NDRangeM = M / myparams::M;  //Assertion would happen at this line
-size_t NDRangeN = N / myparams::N;
 ```
 
-=== Default Values Example:
-```c++
-using myparams = tpu_params_both<tpu::xmx16, int8_t, int8_t, int>;
-// use this to construct the ranges on the host side
-size_t NDRangeM = M / myparams::M;
-size_t NDRangeN = N / myparams::N;
-//if M, N, K do not multiply the default sizes, padding has to be done
-// device code: the matrices are constructed using the default dimensions
-myparams::joint_matrix_a<sub_group, layout::row_major> sub_a;
-myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
-myparams::joint_matrix_accumulator<sub_group> sub_c;
-
-```
-
-=== General Query Example:
+===== General Query Example:
 ```c++
 constexpr int M = 1500; // with msize = 8 and msize = 4,
-          // M can be broken up to 125 sequence of 8-sized ops and remaining 500 using 125 sequence of 4-sized ops
-tpu_params<tpu::xmx16> params;
-constexpr int msize = break_dimension(params, M);
-constexpr int msize_remainder = break_dimension_remainder(params, M);
-constexpr int nsize = params.combinations[0].nsize;
-constexpr int ksize = params.combinations[0].ksize;
+          // M can be broken up to 125 sequence of 8-sized ops and
+          // remaining 500 using 125 sequence of 4-sized ops
+auto combinations = device.get_info<info::device::matrix::combinations>();
+
+constexpr int {msize, nsize, ksize} = break_dimension(combinations, M);
+constexpr int msize_remainder = break_dimension_remainder(combinations, M);
 // device code:
 joint_matrix<sub_group, int8_t, use::a, msize, ksize, layout::row_major> sub_a;
 joint_matrix<sub_group, int8_t, use::b, ksize, nsize, layout::row_major> sub_b;
 joint_matrix<sub_group, int, use::accumulator, msize, nsize> sub_c;
 //Remainder handling
```

Comment:
This is not a realistic example because of all the `constexpr`.  The `combinations` value is not `constexpr`, so the `msize`, `nsize`, and `ksize` can't be `constexpr` either.  Therefore, they cannot be used as template parameters.

Reply:
You are right, I will change that.